os/fs/smartfs/smartfs_smart.c

```diff
@@ -1696,13 +1716,25 @@ static int smartfs_mkdir(struct inode *mountpt, const char *relpath, mode_t mode
 
 		/* Create the directory */
 
-		ret = smartfs_createentry(fs, parentdirsector, filename,
-								  SMARTFS_DIRENT_TYPE_DIR, mode, &entry, 0xFFFF, NULL);
-		if (ret != OK) {
+#ifdef CONFIG_SMARTFS_USE_SECTOR_BUFFER
+		ret = smartfs_find_firstinvalidentry(fs, parentdirsector, &invalidentry);
+		if (ret == OK) {
```

Comment:
Let's clean here.
First we'd better clear specific error first such like a this situation.
```
if (ret != OK)  {
    if (ret != -ENOENT) {
          goto error_with_semaphore;
    } else {
         
    }
}